src/utils/useQueryFilters.ts

```diff
@@ -0,0 +1,129 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import _ from 'lodash';
+import useQuery from 'src/utils/useQuery';
+import useStateLocation, { getLocation } from 'src/utils/useStateLocation';
+import { isArray } from 'src/types/utils';
+
+type FilterValue = (string | number | null)[] | boolean | null;
+export type FiltersType = Record<string, FilterValue>;
+
+// Convert an array to a comma separated string and other values to a string
+// Array values are sorted, so we can detect equality easier elsewhere
+const normalizeValue = (value: FilterValue): string => {
+  if (isArray(value)) {
+    return value
+      .sort((valueA, valueB) => `${valueA}`.localeCompare(`${valueB}`))
+      .map((_value) => `${_value}`)
+      .join(',');
+  } else if (value === null) {
+    return '';
+  } else {
+    return `${value}`;
+  }
+};
+
+const denormalizeNumber = (value: string): number | string =>
+  value.match(/^[0-9]+$/) && `${Number(value)}` === value ? _.toFinite(value) : value;
```

Comment:
do we care about floating point?

Reply:
We may in the future, but as far as I can tell we only have filters that are either a boolean or an array of numbers